src/store/request/store.js

```diff
@@ -94,30 +102,34 @@ class RequestStore extends CoreStore {
     addClearRequestsListener(cb) {
         this.addListener(CLEAR, cb);
     }
+
     /**
    * Remove a listener on the global change on the search store.
    * @param {function} cb - The callback to called when a message is pushed.
    */
     removeClearRequestsListener(cb) {
         this.removeListener(CLEAR, cb);
     }
+
+    /**
+     * The store register itself on the dispatcher.
+     *
+     * @memberof RequestStore
+     */
     registerDispatcher() {
-        let currentStore = this;
-        this.dispatch = AppDispatcher.register(function (transferInfo) {
-            let rawData = transferInfo.action.data;
-            let type = transferInfo.action.type;
-            if (!rawData || !rawData.request) { return; }
+        this.dispatch = AppDispatcher.register(({ action: { data: rawData, type } }) => {
+            if (!rawData || !rawData.request) {
+                return;
+            }
             switch (type) {
                 case 'update':
-                    if (rawData.request) {
```

Comment:
isn't the if statement a fail-safe against a generic `update` generic action type ?

Reply:
Yeah, this is a mistake. I should put it back, or add an identifier.
I cannot think of projects using directly this store with dispatch.